File: WinForms/DevExpress.OutlookInspiredApp.Win/CS/DevExpress.OutlookInspiredApp.Win/ViewModels/Task/TaskViewModel.partial.cs
namespace DevExpress.DevAV.ViewModels {
    using System;

    partial class TaskViewModel : ISupportNewChid {
        public new bool IsNew() { return base.IsNew(); }
        protected override string GetTitle() {
            return Entity.Subject;
        }
        protected override void OnEntityChanged() {
            base.OnEntityChanged();
            EventHandler handler = EntityChanged;
            if(handler != null)
                handler(this, EventArgs.Empty);
        }
        public event EventHandler EntityChanged;
    }
    public partial class SynchronizedTaskViewModel : TaskViewModel {
        protected override bool EnableSelectedItemSynchronization {
            get { return true; }
        }
        protected override bool EnableEntityChangedSynchronization {
            get { return true; }
        }
    }
}

File: WinForms/DevExpress.OutlookInspiredApp.Win/VB/DevExpress.OutlookInspiredApp.Win/ViewModels/Task/TaskViewModel.partial.vb
Imports System

Namespace DevExpress.DevAV.ViewModels

    Partial Class TaskViewModel
        Implements ISupportNewChid

        Public Overloads Function IsNew() As Boolean
            Return MyBase.IsNew()
        End Function

        Protected Overrides Function GetTitle() As String
            Return Entity.Subject
        End Function

        Protected Overrides Sub OnEntityChanged()
            MyBase.OnEntityChanged()
            Dim handler As EventHandler = EntityChangedEvent
            If handler IsNot Nothing Then handler(Me, EventArgs.Empty)
        End Sub

        Public Event EntityChanged As EventHandler
    End Class

    Public Partial Class SynchronizedTaskViewModel
        Inherits TaskViewModel

        Protected Overrides ReadOnly Property EnableSelectedItemSynchronization As Boolean
            Get
                Return True
            End Get
        End Property

        Protected Overrides ReadOnly Property EnableEntityChangedSynchronization As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace
